src/inputs-controller.ts

```diff
@@ -251,7 +253,7 @@ export class InputsController {
     if (gamepadName) {
       this.selectedDevice[Device.GAMEPAD] = gamepadName.toLowerCase();
     }
-    const handler = globalScene.ui?.handlers[UiMode.SETTINGS_GAMEPAD] as SettingsGamepadUiHandler;
+    const handler = globalScene.ui?.handlers[UiMode.SETTINGS_GAMEPAD] as GamepadSettingsUiHandler;
```

Comment:
Q: We should possibly update the `Ui.getHandler<T>()` method to also allow a parameter. Thus we could call
```suggestion
    const handler = globalScene.ui?.getHandler<GamepadSettingsUiHandler>(UiMode.SETTINGS_GAMEPAD);
```
then

Reply:
Definitely if we want to keep things this way yes.
But ideally we should get rid of all those places in the code that access the handlers like this, I don't think they should be accessed publicly like this. I suspect most of the places where this is done would instead be replaced by events
Right now this whole thing only works because all handlers are always in memory, and hopefully the next big UI PR after this one will change that, so this is likely when this issue will be addressed.